core/src/main/java/org/elasticsearch/index/query/NotQueryParser.java

```diff
@@ -56,12 +55,16 @@ public NotQueryBuilder fromXContent(QueryParseContext parseContext) throws IOExc
                 // skip
             } else if (token == XContentParser.Token.START_OBJECT) {
                 if (parseContext.parseFieldMatcher().match(currentFieldName, QUERY_FIELD)) {
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
                     queryFound = true;
                 } else {
                     queryFound = true;
                     // its the filter, and the name is the field
-                    query = parseContext.parseInnerFilterToQueryBuilder(currentFieldName);
+                    QueryParser queryParser = parseContext.queryParser(currentFieldName);
```

Comment:
This was the the only place where parseInnerFilterToQueryBuilder(currentFieldName) was still used, where the current field name actually is the name of the inner query. Seems to be a special case, so I moved the few lines of code here. I gues the query DSL allows for this variant so we can't get rid of it completely yet.